svc-systems/systems/storage.go

```diff
@@ -254,10 +254,108 @@ func searchItem(slice []string, val string) bool {
 	return false
 }
 
-// DeleteVolume is the handler for deleting a volume
+// DeleteVolume defines the logic for deleting a volume under storage
 func (e *ExternalInterface) DeleteVolume(req *systemsproto.VolumeRequest) response.RPC {
-	// TODO add functionality to delete a volume
-	return response.RPC{
-		StatusCode: http.StatusNotImplemented,
+	var resp response.RPC
+
+	var volume smodel.Volume
+	// unmarshalling the volume
+	err := json.Unmarshal(req.RequestBody, &volume)
+	if err != nil {
+		errorMessage := "Error while unmarshaling the create volume request: " + err.Error()
+		log.Printf(errorMessage)
+		resp = common.GeneralError(http.StatusBadRequest, response.MalformedJSON, errorMessage, []interface{}{}, nil)
+		return resp
+	}
+
+	// spliting the uuid and system id
+	requestData := strings.Split(req.SystemID, ":")
```

Comment:
Scenarios for "uuid:" where there is no system id present should be handled as the len(requestData) will be 2

Reply:
Add UT for the same

Reply:
Handled the scenario and added test case for that